devito/passes/iet/parpragma.py

```diff
@@ -172,7 +179,22 @@ def _find_collapsable(self, root, candidates):
                         pass
 
                 collapsable.append(i)
-        return collapsable
+
+            # Give a score to this candidate, based on the number of fully-parallel
+            # Iterations and their position (i.e. outermost to innermost) in the nest
+            score = (
```

Comment:
how does that help?

Reply:
example: 

```
for t
   if t % 4 == 0:
      for x
         for y
             u[t, x, y] = u[t, x, y] + 1
```

in this dummy example, all of the three loops are PARALLEL. But in master the compiler would pick `t`. With this new ranking code it will instead pick the collapsable x-y pair, which has a much greater amount of parallelism. This is especially useful on GPUs...